### ISSUE TITLE
OSError: [Errno 17] File exists: '/home/jlebar/code/moz/ff-git/debug/js/.hacky'

### ISSUE THREAD

**Issue by jlebar** [bgirard/hacky.mk#6]
Got this while doing a top-level clobber make -f client.mk.

```
Traceback (most recent call last):
  File "../../../../src/js/src/tools/hackymake/makehacky.py", line 309, in <module>
    makecchacky(*args[1:])
  File "../../../../src/js/src/tools/hackymake/makehacky.py", line 203, in makecchacky
    hackyfile = openhacky()
  File "../../../../src/js/src/tools/hackymake/makehacky.py", line 117, in openhacky
    if not os.path.exists(hackydir): os.mkdir(hackydir)
OSError: [Errno 17] File exists: '/home/jlebar/code/moz/ff-git/debug/js/.hacky'
```

**Comment by jlebar** [comment]
Oh, here's the line above in the build:

/usr/local/bin/ccache clang -Qunused-arguments -o sysunix.o -c -I../../../dist/system_wrappers_js -include ../../../../src/js/src/config/gcc_hidden.h -DANSI_ARROWS -DHAVE_TCGETATTR -DHIDE -DUSE_DIRENT -DSYS_UNIX -DHAVE_STDLIB -DUNIQUE_HISTORY -DMOZ_GLUE_IN_PROGRAM -DNO_NSPR_10_SUPPORT  -I../../../../src/js/src/editline -I. -I../../../dist/include  -I/home/jlebar/code/moz/ff-git/debug/dist/include/nspr       -fPIC -Qunused-arguments  -Qunused-arguments -Wall -Wpointer-arith -Wdeclaration-after-statement -Werror=return-type -Wtype-limits -Wempty-body -Wsign-compare -Wno-unused -std=gnu99 -fgnu89-inline -ffunction-sections -fdata-sections -pthread -pipe  -DDEBUG -D_DEBUG -DTRACING -g -fno-omit-frame-pointer   -Qunused-arguments  -include ../js-confdefs.h -DMOZILLA_CLIENT -MD -MP -MF .deps/sysunix.o.pp  /home/jlebar/code/moz/ff-git/src/js/src/editline/sysunix.c

**Comment by bgirard** [comment]
That line should be fine assuming there's no race condition with another process 'if not os.path.exists(hackydir): os.mkdir(hackydir)' so that must be it.

**Comment by jlebar** [comment]
Okay, I can try to fix this then.

**Comment by vvuk** [comment]
A "real" fix would be to have mach/configure/whatever just create
.hacky/{ninja,hacky,make} .  A temporary fix would be to just make those
dirs before running a build in your objdir :)

**Comment by bgirard** [comment]
it used to do that in the Makefile but someone patches it out :P

**Comment by vvuk** [comment]
It was in client.mk, who uses client.mk? :p  Merged #7
